### ISSUE TITLE
Compile error on Flutter 1.5.5

### ISSUE THREAD

**Issue by mateusfsilva** [zino-app/graphql-flutter#238]
**Describe the bug**
After update flutter to version 1.5.5 the following error is happening on run:

```
Compiler message:
file:///Users/mateus.gustavo/desenvolvimento/flutter/.pub-cache/hosted/pub.dartlang.org/graphql_flutter-1.0.0+3/lib/src/widgets/subscription.dart:15:3: Error: Function-typed
parameters can't specify 'const', 'final' or 'var' in place of a return type.
Try replacing the keyword with a return type.
  final bool loading,
  ^^^^^
file:///Users/mateus.gustavo/desenvolvimento/flutter/.pub-cache/hosted/pub.dartlang.org/graphql_flutter-1.0.0+3/lib/src/widgets/subscription.dart:16:3: Error: Function-typed
parameters can't specify 'const', 'final' or 'var' in place of a return type.
Try replacing the keyword with a return type.
  final T payload,
  ^^^^^
file:///Users/mateus.gustavo/desenvolvimento/flutter/.pub-cache/hosted/pub.dartlang.org/graphql_flutter-1.0.0+3/lib/src/widgets/subscription.dart:17:3: Error: Function-typed
parameters can't specify 'const', 'final' or 'var' in place of a return type.
Try replacing the keyword with a return type.
  final dynamic error,
  ^^^^^
```

**To Reproduce**
1. `flutter channel dev`
2. `flutter upgrade`
3. `flutter run` or `flutter test`

**Expected behavior**
The project needs to run without compile errors.

**Desktop:**
 - OS: macOS High Sierra
 - Version 10.13.6

**Additional context**
To fix the issue just delete the `final` keyword from the parameters of the `SubscriptionBuilder` typedef.

```dart
typedef SubscriptionBuilder<T> = Widget Function({
  bool loading,
  T payload,
  dynamic error,
});
```

**Comment by psyanite** [comment]
I have the same issue

**Comment by mainawycliffe** [comment]
@psyanite  Are you too using dart 2.3?

**Comment by psyanite** [comment]
Flutter 1.5.5 and 1.5.4 failed. 1.5.3 worked.

**Comment by mainawycliffe** [comment]
It seems this a bug with dart 2.3. @mateusfsilva has a PR on the same, you can try it out.

**Comment by micimize** [comment]
merged and should be fixed in `1.0.0+4` (tested on my own depending app)

It might be a regression in dart 2.3, but it might also have been previously-invalid syntax that the compiler didn't throw on. Regardless, the signature was incorrect

**Comment by psyanite** [comment]
Thanks @micimize and @mainawycliffe.
